source/language/openpulse.rst

```diff
@@ -0,0 +1,650 @@
+    // desired Rabi rate.
+    q1_π_half_sig = mix(π_half_sig, sine(q1_π_half_amp, q1_pos_freq-qubit_freq, 0.0, π_half_time));
+    q2_π_half_sig = mix(π_half_sig, sine(q2_π_half_amp, q2_pos_freq-qubit_freq, 0.0, π_half_time));
+    q3_π_half_sig = mix(π_half_sig, sine(q3_π_half_amp, q3_pos_freq-qubit_freq, 0.0, π_half_time));
+
+    for τ in [0: T]:
+        // Simultaneous π/2 pulses
+        play(eom_a_channel, constant(raman_a_amp, π_half_time) , raman_a_frame);
+        play(eom_b_channel, constant(raman_b_amp, π_half_time) , raman_b_frame);
+        play(aod_channel, q1_π_half_sig, q1_frame);
+        play(aod_channel, q1_π_half_sig, q2_frame);
+        play(aod_channel, q1_π_half_sig, q3_frame);
+
+        // Time delay all
+        delay(τ/2)
+
+        // π pulse on qubit 1 only -- composed of two π/2 pulses
+        for _ in [0:1]:
+            play(eom_a_channel, constant(raman_a_amp, π_half_time) , raman_a_frame);
+            play(eom_b_channel, constant(raman_b_amp, π_half_time) , raman_b_frame);
+            play(aod_channel, q1_π_half_sig, q1_frame);
+
+        // Barrier all then time delay all
+        barrier()
+        delay(τ/2)
+
+        // Phase shift the signals by a different amount -- or should I be shifting qubit_#_signal?
+        q1_frame.phase += tppi_1 * τ
+        q1_frame.phase += tppi_2 * τ
+        q1_frame.phase += tppi_3 * τ
+
+        // Simultaneous π/2 pulses
+        play(eom_a_channel, constant(raman_a_amp, π_half_time) , raman_a_frame);
+        play(eom_b_channel, constant(raman_b_amp, π_half_time) , raman_b_frame);
+        play(aod_channel, q1_π_half_sig, q1_frame);
+        play(aod_channel, q1_π_half_sig, q2_frame);
+        play(aod_channel, q1_π_half_sig, q3_frame);
+  }
+
+Multiplexed readout and capture
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, we want to perform readout and capture of a pair of qubits, but mediated by a
+single physical tx and rx channel. The example is for just two qubits, but works the same for
+many (just adding more frames, waveforms, plays, and captures).
+
+.. code-block:: javascript
+
+  defcal multiplexed_readout_and_capture $0 $1 {
+
+      // the tx/rx channel is the same for $0 and $1
+      tx_channel ro_tx = txch($0, "readout");
+      rx_channel ro_rx = rxch($0, "readout");
+
+      // readout frames of different frequencies
+      q0_frame = Frame(q0_ro_freq, 0); // time 0
+      q1_frame = Frame(q1_ro_freq, 0); // time 0
+
+      // flat-top readout waveforms
+      waveform q0_ro_wf = constant(amp=0.1, l=...);
+      waveform q1_ro_wf = constant(amp=0.2, l=...);
+
+      // multiplexed readout
+      play(ro_tx, q0_ro_wf, q0_frame);
+      play(ro_tx, q1_ro_wf, q1_frame);
+
+      // simple boxcar kernel
+      waveform ro_kernel = constant(amp=1, l=...);
+
+      // multiplexed capture
+      // kernel capture(channel chan, waveform ro_kernel, frame capture_frame) -> bit;
+      bit q0_bit = capture(ro_rx, ro_kernel, q0_frame);
```

Comment:
I don't think we can reuse the frame for `capture` because the `capture` would be scheduled after the `play`.

Reply:
Yes, I think we need to split this out into stimulus and capture frames. 
```
// readout stimulus and capture frames of different frequencies
frame q0_stimulus_frame = newframe(q0_ro_freq, 0); // time 0
frame q0_capture_frame = newframe(q0_ro_freq, 0); // time 0
frame q1_stimulus_frame = newframe(q1_ro_freq, 0); // time 0
frame q1_capture_frame = newframe(q1_ro_freq, 0); // time 0
```

Then we would want to use a `barrier` before capturing on the `q#_capture_frame`s